modules/ensemble/lib/action/invoke_api_action.dart

```diff
@@ -144,13 +145,27 @@ class InvokeAPIController {
         }
 
         APIProvider apiProvider = getAPIProvider(context, apiDefinition);
+        var isChangeListener = apiDefinition['listenForChanges'];
+
+        if (isChangeListener is! bool) {
+          isChangeListener = apiScopeManager.dataContext
+              .getContextById(apiDefinition['listenForChanges']);
+
+          if (!isChangeListener) {
+            if (apiProvider is LiveAPIProvider) {
+              await (apiProvider as LiveAPIProvider)
+                  .unSubscribeToApi(action.apiName);
```

Comment:
we need to consider the API's with the same name but different IDs. The use case there is calling the same API with different inputs and using the API's id to bind it to different widgets. This code will cancel subscriptions for any existing APIs with the same name but different ids

Reply:
I agree. I'll implement once I prioritise it.